server/src/main/java/io/whitefox/services/DeltaSharesServiceImpl.java

```diff
@@ -52,6 +54,15 @@ public CompletionStage<Optional<Share>> getShare(String share) {
         .thenApplyAsync(o -> o.map(DeltaSharesServiceImpl::pShareToShare));
   }
 
+  @Override
+  public CompletionStage<Optional<Integer>> getTableVersion(
+      String share, String schema, String table, Optional<String> startingTimestamp) {
+    var tableRes = storageManager.getTable(share, schema, table);
+    return tableRes.thenApply(t -> t.map(DeltaSharedTable::new)).thenCompose(odt -> odt.map(
+            dt -> dt.getTableVersion(startingTimestamp).thenApply(Optional::ofNullable))
```

Comment:
I think that if the table exists, and the snapshot exists, then the version is not optional